src/currency/currency-formatter.ts

```
@@ -0,0 +1,72 @@
+import { CurrencyConfig } from './currency';
+
+export default class CurrencyFormatter {
+    private _decimalPlaces: number;
+    private _symbolLocation: string;
+    private _symbol: string;
+    private _thousandsSeparator: string;
+    private _decimalSeparator: string;
+
+    constructor(
+        currencySettings?: CurrencyConfig | any
+    ) {
+        if (!currencySettings) {
+            throw new Error('Currency settings missing');
+        }
+
+        const {
+            decimalPlaces,
+            symbolLocation,
+            symbol,
+            thousandsSeparator,
+            decimalSeparator,
+        } = currencySettings;
+
+        if (
+            typeof symbolLocation !== 'string' ||
+            typeof symbol !== 'string' ||
+            typeof thousandsSeparator !== 'string' ||
+            typeof decimalSeparator !== 'string' ||
+            typeof decimalPlaces !== 'string'
+        ) {
+            throw new Error('Invalid currency settings provided');
+        }
+
+        this._decimalPlaces = parseInt(decimalPlaces, 10);
+        this._symbolLocation = symbolLocation;
+        this._symbol = symbol;
+        this._thousandsSeparator = thousandsSeparator;
+        this._decimalSeparator = decimalSeparator;
+    }
+
+    format(amount?: number): string {
+        if (typeof amount !== 'number') {
+            throw new Error('Invalid amount provided');
```

Maybe throw `InvalidArgument` instead?